test/entrypoint.test.ts

```diff
@@ -427,6 +429,25 @@ describe('EntryPoint', function () {
         expect(e.message).to.include('Revert after first validation')
       }
     })
+
+    it('should limit revert reason length before emitting it', async () => {
+      const REVERT_REASON_MAX_LEN = 1e5
+      const testRevertAccount = await new TestRevertAccount__factory(ethersSigner).deploy(entryPoint.address, { value: '0x' + 1e18.toString(16) })
+      const badData = await testRevertAccount.populateTransaction.revertLong(REVERT_REASON_MAX_LEN + 1)
+      const badOp: UserOperation = {
+        ...DefaultsForUserOp,
+        sender: testRevertAccount.address,
+        callGasLimit: 1e5,
+        maxFeePerGas: 1,
+        verificationGasLimit: 1e5,
+        callData: badData.data!
+      }
+      const beneficiaryAddress = createAddress()
+      await expect(entryPoint.simulateValidation(badOp, { gasLimit: 1e6 }))
+        .to.revertedWith('ValidationResult')
+      const tx = await entryPoint.handleOps([badOp], beneficiaryAddress, { gasLimit: 1e6 })
```

Comment:
1. you test it with huge gaslimit: the whole idea is that it doesn't take too much gas, but you check with 10 times gasLimit. should test with (say) callGas + verification + 10k
2. the test should follow its title, and check the returned reason is up to 2k bytes.